homeassistant/components/electrolux_group/__init__.py

```diff
@@ -0,0 +1,217 @@
+    ConfigEntryNotReady,
+    ServiceValidationError,
+)
+from homeassistant.helpers import (
+    config_validation as cv,
+    device_registry as dr,
+    entity_registry as er,
+)
+from homeassistant.helpers.typing import ConfigType
+
+from .api import ElectroluxApiClient
+from .const import CONF_REFRESH_TOKEN, DOMAIN, USER_AGENT
+from .coordinator import ElectroluxDataUpdateCoordinator
+
+CONFIG_SCHEMA = cv.config_entry_only_config_schema(DOMAIN)
+
+_LOGGER: logging.Logger = logging.getLogger(__package__)
+
+PLATFORMS = [
+    Platform.SENSOR,
+]
+
+
+@dataclass(kw_only=True, slots=True)
+class ElectroluxData:
+    """Electrolux data type."""
+
+    client: ElectroluxApiClient
+    coordinators: dict[str, ElectroluxDataUpdateCoordinator]
+    sse_task: Task
+
+
+type ElectroluxConfigEntry = ConfigEntry[ElectroluxData]
+
+
+async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
+    """Set up Electrolux Group integration."""
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
+    """Set up Electrolux Group integration entry."""
+
+    def save_tokens(new_access, new_refresh, api_key):
+        hass.config_entries.async_update_entry(
+            entry,
+            data={
+                **entry.data,
+                CONF_API_KEY: api_key,
+                CONF_ACCESS_TOKEN: new_access,
+                CONF_REFRESH_TOKEN: new_refresh,
+            },
+        )
+
+    try:
+        token_manager = ElectroluxTokenManager(hass, entry, save_tokens)
+        appliance_client = ApplianceClient(
+            token_manager=token_manager, external_user_agent=USER_AGENT
+        )
+    except Exception as exception:
+        raise ConfigEntryAuthFailed("Failed to setup API client") from exception
+
+    client = ElectroluxApiClient(appliance_client)
+
+    # Check during integration initialization if we are able to set it up correctly
+    try:
+        await appliance_client.test_connection()
+    except FailedConnectionException as e:
+        raise ConfigEntryAuthFailed("Connection with client failed.") from e
+
+    appliances = await client.fetch_appliance_data()
+
+    coordinators: dict[str, ElectroluxDataUpdateCoordinator] = {}
+    on_livestream_opening_callback_list: list[Callable[[], Awaitable[None]]] = []
+
+    async def check_for_new_devices_callback() -> None:
+        """Trigger _check_for_new_devices asynchronously."""
+        await _check_for_new_devices(hass, entry, client)
+
+    on_livestream_opening_callback_list.append(check_for_new_devices_callback)
+
+    for appliance in appliances:
+        appliance_id = appliance.appliance.applianceId
+        coordinator = ElectroluxDataUpdateCoordinator(
+            hass, entry, client=client, applianceId=appliance_id
+        )
+
+        await coordinator.async_config_entry_first_refresh()
+
+        if not coordinator.last_update_success:
+            raise ConfigEntryNotReady from coordinator.last_exception
```

Comment:
if `_async_update_data` raised `UpdateFailed`, it would've raised `ConfigEntryNotReady` already